File: JG/update_html.js
const fs = require('fs');

let html = fs.readFileSync('index.html', 'utf8');

const row1 = `
                            <div class="fold-marquee">
                                <div class="fold-track">
                                    <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits.
                                </div>
                            </div>`;
const row2 = `
                            <div class="fold-marquee">
                                <div class="fold-track">
                                    Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits.
                                </div>
                            </div>`;
const row3 = `
                            <div class="fold-marquee">
                                <div class="fold-track">
                                    Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span>
                                </div>
                            </div>`;
const row4 = `
                            <div class="fold-marquee">
                                <div class="fold-track">
                                    <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits.
                                </div>
                            </div>`;

const newContent = `${row1}${row2}${row3}${row4}\n                        `;

// Replace contents of all three `.fold-content` div elements securely
// Find the index of <div class="fold-content"
let searchIndex = 0;
while (true) {
    let startIdx = html.indexOf('<div class="fold-content"', searchIndex);
    if (startIdx === -1) break;
    let endOfStartTag = html.indexOf('>', startIdx) + 1;
    
    // Find the matching closing </div>
    // Since there are 4 inner fold-marquee divs, we can just search for the first </div> that closes fold-content
    // Actually, each fold-marquee has 2 closing divs. Total 8 inner closing divs.
    // Let's just find the exact string we know is at the end of fold-content.
    // We can use a regex that matches until `</div>\n                    </div>\n                </div>`
    
    // Instead, let's use a simple regex on the substring.
    let remainder = html.substring(endOfStartTag);
    // Find the sequence of closing tags for fold-content, fold-align, fold...
    let closeIdx = remainder.indexOf('</div>\\n                    </div>\\n                </div>');
    if (closeIdx === -1) {
       closeIdx = remainder.indexOf('</div>\\r\\n                    </div>\\r\\n                </div>');
    }
    if (closeIdx === -1) {
       // fallback: just find "</div>" followed by whitespace and "</div>"
       let match = remainder.match(/<\\/div>\\s*<\\/div>\\s*<\\/div>/);
       if (match) {
           closeIdx = match.index;
       }
    }
}

File: JG/update_html.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    html = f.read()

# Let's just find the fold-marquee divs inside the fold-content by matching the EXACT content that was previously there.
# Since it's exactly the same in all three places, we can just replace the 4 marquees.

old_marquee = '''<div class="fold-marquee">
                                <div class="fold-track">
                                    Create beyond Limits. Create beyond Limits. <span class="fold-focus">Create beyond
                                        Limits.</span> Create beyond Limits. Create beyond Limits. Create beyond Limits.
                                    Create beyond Limits. <span class="fold-focus">Create beyond Limits.</span> Create
                                    beyond Limits.
                                </div>
                            </div>'''

# Clean up whitespace differences by using regex to match the pattern
pattern = r'<div class="fold-marquee">\s*<div class="fold-track">\s*Create beyond Limits\.\s*Create beyond Limits\.\s*<span class="fold-focus">Create beyond\s*Limits\.<\/span>\s*Create beyond Limits\.\s*Create beyond Limits\.\s*Create beyond Limits\.\s*Create beyond Limits\.\s*<span class="fold-focus">Create beyond Limits\.<\/span>\s*Create\s*beyond Limits\.\s*<\/div>\s*<\/div>'

row1 = '''                            <div class="fold-marquee">
                                <div class="fold-track">
                                    <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits.
                                </div>
                            </div>'''
row2 = '''\n                            <div class="fold-marquee">
                                <div class="fold-track">
                                    Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits. Create <span class="fold-focus">beyond</span> Limits.
                                </div>
                            </div>'''
row3 = '''\n                            <div class="fold-marquee">
                                <div class="fold-track">
                                    Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span> Create beyond <span class="fold-focus">Limits.</span>
                                </div>
                            </div>'''
row4 = '''\n                            <div class="fold-marquee">
                                <div class="fold-track">
                                    <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits. <span class="fold-focus">Create</span> beyond Limits.
                                </div>
                            </div>'''

new_content = row1 + row2 + row3 + row4

# We have 4 identical old_marquees inside each fold-content, repeated 3 times = 12 total old_marquees.
# But we want to replace each block of 4 old_marquees with 1 block of new_content!

# Wait, if we replace 4 old marquees with 4 new marquees, we can just find the block of 4!
full_pattern = pattern + r'\s*' + pattern + r'\s*' + pattern + r'\s*' + pattern

html = re.sub(full_pattern, new_content, html, flags=re.DOTALL)

with open('index.html', 'w', encoding='utf-8') as fw:
    fw.write(html)
print('Replaced successfully')
